src/commands/pull.js

```diff
@@ -0,0 +1,157 @@
+const fs = require('fs');
+const path = require('path');
+const tiged = require('tiged');
+const AdmZip = require('adm-zip');
+const semverGt = require('semver/functions/gt');
+const { get } = require('axios');
+const { flags: flagsHelper } = require('@oclif/command');
+const jju = require('jju');
+const { ApifyCommand } = require('../lib/apify_command');
+const { success, error } = require('../lib/outputs');
+const { getLoggedClientOrThrow, getLocalConfigOrThrow, getLocalUserInfo } = require('../lib/utils');
+const { LOCAL_CONFIG_PATH } = require('../lib/consts');
+
+const extractGitHubZip = async (url, directoryPath) => {
+    const { data } = await get(url, { responseType: 'arraybuffer' });
+
+    const zipFile = new AdmZip(Buffer.from(data, 'binary'));
+
+    zipFile.extractEntryTo(zipFile.getEntries()[0].entryName, directoryPath, false);
+};
+
+class PullCommand extends ApifyCommand {
+    async run() {
+        const { args, flags } = this.parse(PullCommand);
+        const localConfig = await getLocalConfigOrThrow();
+        const userInfo = await getLocalUserInfo();
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const isActorAutomaticallyDetected = !args?.actorId;
+        const usernameOrId = userInfo.username || userInfo.id;
+
+        const actorId = args?.actorId || localConfig?.id || (localConfig.name ? `${usernameOrId}/${localConfig.name}` : undefined);
+
+        if (!actorId) throw new Error('Cannot find Actor in this directory.');
+
+        let actor;
+        try {
+            actor = await apifyClient.actor(actorId).get();
+        } catch {
+            throw new Error(`Cannot find Actor with ID/name '${actorId}' in your account.`);
+        }
+
+        if (!actor) throw new Error(`Cannot find Actor with ID/name '${actorId}' in your account.`);
+
+        const { name, versions } = actor;
+
+        let correctVersion = null;
+        if (flags?.version) {
+            correctVersion = versions.find((version) => version.versionNumber === flags?.version);
+            if (!correctVersion) {
+                error(`Cannot find version ${flags?.version} of Actor ${actorId}. Pulling latest version instead.`);
```

Comment:
I think if a version was specified and it can't be found, you should throw an error instead of pulling the latest version.

If I did `git merge my-feature` and it said `can't find branch my-feature, merging "master" instead`, I wouldn't be very happy.